tools/webassembly.py

```diff
@@ -115,29 +108,53 @@ def add_emscripten_metadata(wasm_file):
     f.write(orig[8:])
 
 
+class Module:
+  """Extremely minimal wasm module reader.  Currently only used
+  for parsing the dylink section."""
```

Comment:
is there a goal to use it for anything else? (I thought the one other use case, for EM_ASMs, went away)

Reply:
I think we can/should still look into reading all metadata using this code.   There is a minor difficulty due to active segments but I believe we can find a way around that.

Reply:
Why? I thought we'd agreed parsing it in binaryen is simpler and faster (now that we skip functions and names).

Reply:
I guess I hadn't fully understood that conclusion.   I'm on board with the idea of keeping binaryen around for metadata parsing if it really is faster.   But the idea of completely removing the brittle ABI between emscripten and binaryen is still somewhat alluring to me.    I agree its not worth investing in right now given all our other priorities though.

Reply:
Ah I forgot, I'm also planning to use this code to read the imports from shared libraries as part of #13830

Reply:
Oh, that's an interesting use case. Let's discuss further there.